src/Paprika/Store/PagedDb.MultiHeadChain.cs

```diff
@@ -0,0 +1,977 @@
+﻿using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading.Channels;
+using Paprika.Chain;
+using Paprika.Crypto;
+using Paprika.Data;
+using Paprika.Utils;
+
+namespace Paprika.Store;
+
+public sealed partial class PagedDb
+{
+    public IMultiHeadChain OpenMultiHeadChain()
+    {
+        lock (_batchLock)
+        {
+            if (_batchCurrent != null)
+            {
+                ThrowOnlyOneBatch();
+            }
+
+            var chain = new MultiHeadChain(this);
+            _batchCurrent = chain;
+
+            return chain;
+        }
+    }
+
+    private class MultiHeadChain : IMultiHeadChain
```

Comment:
It would be good to add some descriptive comment here to summarize this class/approach?
In general, some of the classes/interfaces in this file have their description missing.